packages/nodes-base/nodes/Evaluation/utils/evaluationUtils.ts

```diff
@@ -96,33 +131,86 @@ export async function setOutputs(this: IExecuteFunctions): Promise<INodeExecutio
 		}
 	});
 
-	await googleSheetInstance.updateRows(
-		googleSheet.title,
-		[columnNames],
-		'RAW', // default value for Value Input Mode
-		1, // header row
-	);
-
 	const outputs = outputFields.reduce<JsonObject>((acc, { outputName, outputValue }) => {
 		acc[outputName] = outputValue;
 		return acc;
 	}, {});
 
-	const preparedData = googleSheetInstance.prepareDataForUpdatingByRowNumber(
-		[
-			{
-				row_number: rowNumber,
-				...outputs,
+	const source = this.getNodeParameter('source', 0) as string;
+
+	if (source === 'dataTable') {
+		if (this.helpers.getDataStoreProxy === undefined) {
+			throw new NodeOperationError(
+				this.getNode(),
+				'Attempted to use Data table node but the module is disabled',
+			);
+		}
+
+		const dataTableId = this.getNodeParameter('dataTableId', 0, undefined, {
+			extractValue: true,
+		}) as string;
+		const dataTableProxy = await this.helpers.getDataStoreProxy(dataTableId);
+
+		const rowId = typeof evaluationTrigger.row_id === 'number' ? evaluationTrigger.row_id : 1;
+
+		const data = Object.fromEntries(
+			Object.entries(outputs).map(([k, v]) => [k, toDataTableValue(v)]),
+		);
+
+		const columns = await dataTableProxy.getColumns();
+		for (const [columnName, value] of Object.entries(outputs)) {
+			if (columns.find((c) => c.name === columnName)) {
+				continue;
+			}
+
+			await dataTableProxy.addColumn({
```

Comment:
User may not be aware of the limitations on column name here. If they are used to google sheets, or porting evals from google sheets, the name will not be valid for data store. ex: spaces in the name
In this case, we show a generic "bad column name" message. 
Can we show a more meaningful message when the source is data table. "Column names must begin with a letter and can only include letters, numbers, or underscores"

Reply:
It's a nitpick. If this is easy to get done, let's do it. Othewise, let's create a follow up task for later.

Reply:
Improved the validation handling, now we pass the code through the same validation error and use the same error message even if we hit this check.